app/src/main/java/jp/co/c_lis/ccl/morelocale/ui/locale_list/LocaleListViewModel.kt

```diff
@@ -60,6 +64,8 @@ class LocaleListViewModel @Inject constructor(
 
     fun setLocale(localeItem: LocaleItem) {
         viewModelScope.launch {
+            MoreLocale.setLocale(localeItem.locale)
+
             preferenceRepository.saveLocale(localeItem)
         }
```

Comment:
If the app doesn't have permission. `MoreLocal.setLocale` method throws `InvocationTargetException`.
The caller cannot catch the exception outside CoroutineScope.

```
5279-5279/jp.co.c_lis.ccl.morelocale E/AndroidRuntime: FATAL EXCEPTION: main
    Process: jp.co.c_lis.ccl.morelocale, PID: 5279
    java.lang.reflect.InvocationTargetException
        at java.lang.reflect.Method.invoke(Native Method)
        at jp.co.c_lis.morelocale.MoreLocale.setLocaleList(MoreLocale.java:118)
        at jp.co.c_lis.morelocale.MoreLocale.setLocale(MoreLocale.java:87)
        at jp.co.c_lis.ccl.morelocale.ui.locale_list.LocaleListViewModel$setLocale$1.invokeSuspend(LocaleListViewModel.kt:67)
```

I think it is a good idea to put `MoreLocale.setLocale` inside the ViewModel!

And could you consider that call `MoreLocale.setLocale` from outside `viewModelScope`. Or to respond even more better, I think there is a way to add `MutableLiveData` as  `eventPermissionRequired` to `LocaleListViewModel`.

```suggestion
        MoreLocale.setLocale(localeItem.locale)

        viewModelScope.launch {
            preferenceRepository.saveLocale(localeItem)
        }
```